README.md

```diff
@@ -1,2 +1,168 @@
-# ENGAGE-HF-Firebase
+# ENGAGE-HF Firebase
+
 Firebase cloud hosting infrastructure for the ENGAGE-HF project.
+
+# Data Scheme
+
+This document describes how data is stored in Firestore for the Engage-HF app.
+
+## Custom Types
+
+The following section describes custom types defined for this system to be later used
+
+### LocalizedText
+
+A LocalizedText object shall be used whenever text requires localization or may require in the future. The object may either simply be a single string (in the case of localizations not being available yet) or a dictionary with string keys and string values. If LocalizedText is represented by a string-string dictionary, the keys represent language-codes following the ISO 639-1 (or if necessary ISO 639-1) standard and the respective text in the given language as value.
+
+A LocalizedText object cannot be used in FHIR-conforming types due to its incompatibility with the standard. FHIR types commonly contain text in one language only to be specified using the `language` property.
+
+## /medications
+
+In this section, we describe information regarding all the medications to be specified in the Engage-HF context. These medications may be used by a clinician for medication requests to a patient (/users/$userId$/medicationRequests) or contra-indications (/users/$userId$/allergyIntolerances).
+
+### /medications/$medicationId$
+
+Based on [FHIR Medication](https://hl7.org/fhir/R4B/medication.html):
+
+|Property|Type|Values|Comments|
+|-|-|-|-|
+|id|string|-|[Resource](https://hl7.org/fhir/R4B/resource.html): Logical id of this artifact|
+|meta|Meta|-|[Resource](https://hl7.org/fhir/R4B/resource.html#Meta): Metadata about the resource|
+|identifier|list of Identifier|-|Business identifier for this medication|
+|extension|list of Extension|-|See /medications/$medicationId$/extension for possible values|
+
+Technically, we could use [FHIR MedicationKnowledge](https://hl7.org/fhir/R4B/medicationknowledge.html), but it doesn't seem to be ready for use - or at least take over the `medicineClassification` property from it.
+
+### /medications/$medicationId$/extension
```

Comment:
it could possibly be called `extensions` here and it might also make sense to have interfaces such as:
`/medications/extensions/{extensionId}`
but it probably depends on the visualisation whether this is needed or possibly superfluous

Reply:
Meant to simply specify different values for the extension property defined by FHIR, so we cannot really rename it - but I can rewrite this part to be more specific about this